cpp/arcticdb/processing/clause.cpp

```diff
@@ -739,9 +739,12 @@ std::vector<EntityId> ResampleClause<closed_boundary>::process(std::vector<Entit
     internal::check<ErrorCode::E_ASSERTION_FAILURE>(is_time_type(first_row_slice_index_col.type().data_type()),
                                                     "Cannot resample data with index column of non-timestamp type");
     auto first_ts = first_row_slice_index_col.scalar_at<timestamp>(0).value();
-    // We can use the last timestamp from the first row-slice's index column, as by construction (structure_for_processing) the bucket covering
-    // this value will cover the remaining index values this call is responsible for
-    auto last_ts = first_row_slice_index_col.scalar_at<timestamp>(first_row_slice_index_col.row_count() - 1).value();
+    // If there is only one row slice, then the last index value of interest is just the last index value for this row
+    // slice. If there is more than one, then the first index value from the second row slice must be used to calculate
+    // the buckets of interest, due to an old bug in update. See test_compatibility.py::test_compat_resample_updated_data
```

Comment:
I think you should describe, here or in the test, exactly what the old bug in update was